### ISSUE TITLE
Django 1.8 support

### ISSUE THREAD

**Issue by bikeshedder** [applegrew/django-select2#153]
Since Django 1.8 is going to be released soon I decided to upgrade one of my projects and found that django-select2 no longer works:

```
  File "django_select2/__init__.py", line 111, in <module>
    from .fields import Select2ChoiceField, Select2MultipleChoiceField, HeavySelect2ChoiceField, \
  File "django_select2/fields.py", line 812, in <module>
    class AutoModelSelect2Field(ModelResultJsonMixin, AutoViewFieldMixin, HeavyModelSelect2ChoiceField):
  File "django_select2/fields.py", line 301, in __new__
    return type.__new__(cls, name, bases, dct)
TypeError: Error when calling the metaclass bases
    metaclass conflict: the metaclass of a derived class must be a (non-strict) subclass of the metaclasses of all its bases
```

For the time being I'll stick with Django 1.7. It would be nice if django-select2 would get rid of the "pretty nasty hacky stuff". Those are not my words. I just quoted the comment in the code. ;)

**Comment by codingjoe** [comment]
Well I've seen this in the travis tests for 1.8. I'll investigate. But we've just fixed a major bug for django 1.7 yesterday. I don't really understand the hurry ;)

**Comment by titusz** [comment]
I did some investigation... Unfortunately all the inheritance and metaclass magic is way above my head. But i have a hunch that the error is due to the fact that django added some of its own metaclass magic to the `Field` class: https://github.com/django/django/blame/stable/1.8.x/django/forms/fields.py#L57

**Comment by hrishiballal** [comment]
+1 for this, encountering the same error

**Comment by codingjoe** [comment]
fixed in #160
